src/core/vectortile/qgsvectortileloader.cpp

```diff
@@ -130,6 +131,34 @@ void QgsVectorTileLoader::tileReplyFinished()
   }
   else
   {
+    QString errorMessage;
+    QString errorSource = mUri;
+
+    bool notifyError = false;
+
+    switch ( reply->error() )
+    {
+      case QNetworkReply::OperationCanceledError:
+        break;
+      case QNetworkReply::TimeoutError:
+        break;
+      case QNetworkReply::ContentAccessDenied:
+        notifyError = true;
+        errorMessage = tr( "Access denied to \"%1\"" ).arg( mUri );
+        if ( !reply->data().isEmpty() )
+          errorMessage.append( QStringLiteral( "<br>" ) + reply->data() );
+        break;
+      case QNetworkReply::ContentNotFoundError:
+        errorMessage = tr( "Nothing found for %1" ).arg( mUri );
+        notifyError = true;
+        break;
+      default:
+        break;
+    }
+
+    if ( notifyError )
+      QgsNetworkAccessManager::instance()->notifyError( errorSource, errorMessage );
```

Comment:
Hmm I am not sure if I like the idea that the errors go through NAM... there may be other errors that are not strictly related to network access (e.g. if we fail to decode tiles), so it would be good not to tie error reporting to NAM.

Why not propagate the errors to QgsMapRendererJob::errors() and then have QgisApp show errors in message bar based on those?